tests/e2e/question.spec.js

```diff
@@ -14,51 +14,129 @@
+    expect(await questionPage.isPythonPrimerLinkDisplayed()).toBe(true);
+    // About link.
+    expect(await questionPage.isAboutLinkDisplayed()).toBe(true);
+    // Privacy link.
+    expect(await questionPage.isPrivacyLinkDisplayed()).toBe(true);
+    // Leave Feedback link.
+    expect(await questionPage.isFeedbackLinkDisplayed()).toBe(true);
   });
+
+  it('should display question and coding UIs in 2 columns on large screens',
+     async function() {
+      await testUtils.setLargeScreen();
+
+      let questionUiLocation = await questionPage.getQuestionUiLocation();
+      let questionUiSize = await questionPage.getQuestionUiSize();
+      let codingUiLocation = await questionPage.getCodingUiLocation();
+
+      // Coding and Question UIs should be horizontally aligned.
+      expect(codingUiLocation.y).toEqual(questionUiLocation.y);
+
+      // Coding UI should be to the right of Question UI.
+      expect(codingUiLocation.x).toBeGreaterThan(
+        questionUiLocation.x + questionUiSize.width);
+     });
+
+  it('should display question and coding UIs in 2 rows on small screens',
+     async function() {
+       await testUtils.setSmallScreen();
+
+       let questionUiLocation = await questionPage.getQuestionUiLocation();
+       let codingUiLocation = await questionPage.getCodingUiLocation();
+       let codingUiSize = await questionPage.getCodingUiSize();
+
+       // Coding and Question UI should be vertically aligned.
+       expect(codingUiLocation.x).toEqual(questionUiLocation.x);
+
+       // Question UI should be below Coding UI.
+       expect(questionUiLocation.y).toBeGreaterThan(
+         codingUiLocation.y + codingUiSize.height);
+     });
+
+  it('should fit the question and coding UIs in page width on large screens',
+     async function() {
+      await testUtils.setLargeScreen();
+
+      let windowSize = await testUtils.getWindowSize();
+
+      let codingUiLocation = await questionPage.getCodingUiLocation();
+      let codingUiSize = await questionPage.getCodingUiSize();
+
+      // The right edge of the coding UI should be less than the window width.
+      expect(codingUiLocation.x + codingUiSize.width).toBeLessThan(
+        windowSize.width);
+     });
+
+  it('should fit the question and coding UIs in page width on small screens',
+     async function() {
+      await testUtils.setSmallScreen();
+
+      let windowSize = await testUtils.getWindowSize();
+
+      let codingUiLocation = await questionPage.getCodingUiLocation();
+      let codingUiSize = await questionPage.getCodingUiSize();
+
+      let questionUiLocation = await questionPage.getQuestionUiLocation();
+      let questionUiSize = await questionPage.getQuestionUiSize();
+
+      // The right edge of the question UI should be less than the window width.
+      expect(questionUiLocation.x + questionUiSize.width).toBeLessThan(
+        windowSize.width);
+
+      // The right edge of the coding UI should be less than the window width.
```

Comment:
These tests are great, btw :) I didn't know before that it was possible to check this sort of thing in Protractor. Thanks!

Reply:
They are also fun to observe when running the tests locally!